### PR TITLE
Fix handling 'auto' checks in absolute layout

### PR DESCRIPTION
**Pull request by coado**
## Summary:

Regarding [issue](https://github.com/facebook/react-native/issues/45817) with incorrect layout when `left` is set to `auto`. This PR introduces handling `auto` whenever inline or flex position is checked to be defined and it fixes above issue.

## Tests:
 I have run the provided unit tests and everything passes.

### CODE REVIEW

**Comment by NickGerleman** [comment]
Looks like the JavaScript changes do not compile.

File: javascript/src/Node.h [hunk 1]
```diff
@@ -76,6 +76,7 @@ class Node {
   void setPositionType(int positionType);
   void setPosition(int edge, double position);
   void setPositionPercent(int edge, double position);
+  void Node::setPositionAuto(int edge);
```
Comment (NickGerleman) [inline javascript/src/Node.h:79]:
```suggestion
  void setPositionAuto(int edge);
```

**Comment by NickGerleman** [comment]
Yay, that fixed the JavaScript build. But tests in JavaScript fail with an exception, indicating something isn't being bound correctly (around Node or embind). https://github.com/facebook/yoga/actions/runs/10521212923/job/29151469086?pr=1689

```
Summary of all failing tests
 FAIL  tests/generated/YGAbsolutePositionTest.test.ts
  ● absolute_layout_width_height_left_auto_right

    TypeError: root_child0.setPositionAuto is not a function

      88 |     const root_child0 = Yoga.Node.create(config);
      89 |     root_child0.setPositionType(PositionType.Absolute);
    > 90 |     root_child0.setPositionAuto(Edge.Left);
         |                 ^
      91 |     root_child0.setPosition(Edge.Right, 10);
      92 |     root_child0.setWidth(10);
      93 |     root_child0.setHeight(10);

      at Object.setPositionAuto (tests/generated/YGAbsolutePositionTest.test.ts:90:17)
```

**Comment by coado** [comment]
> Yay, that fixed the JavaScript build. But tests in JavaScript fail with an exception, indicating something isn't being bound correctly. https://github.com/facebook/yoga/actions/runs/10521212923/job/29151469086?pr=1689
> 
> ```
> Summary of all failing tests
>  FAIL  tests/generated/YGAbsolutePositionTest.test.ts
>   ● absolute_layout_width_height_left_auto_right
> 
>     TypeError: root_child0.setPositionAuto is not a function
> 
>       88 |     const root_child0 = Yoga.Node.create(config);
>       89 |     root_child0.setPositionType(PositionType.Absolute);
>     > 90 |     root_child0.setPositionAuto(Edge.Left);
>          |                 ^
>       91 |     root_child0.setPosition(Edge.Right, 10);
>       92 |     root_child0.setWidth(10);
>       93 |     root_child0.setHeight(10);
> 
>       at Object.setPositionAuto (tests/generated/YGAbsolutePositionTest.test.ts:90:17)
> ```

Thanks! I pushed change that should fix JS tests

File: java/tests/generated/com/facebook/yoga/YGAbsolutePositionTest.java [hunk 2]
```diff
@@ -70,6 +70,135 @@ public void test_absolute_layout_width_height_start_top() {
...
+    YogaConfig config = YogaConfigFactory.create();
+
+    final YogaNode root = createNode(config);
+    root.setPositionType(YogaPositionType.ABSOLUTE);
+    root.setWidth(100f);
+    root.setHeight(100f);
+
+    final YogaNode root_child0 = createNode(config);
+    root_child0.setPositionType(YogaPositionType.ABSOLUTE);
+    root_child0.setPositionAuto(YogaEdge.LEFT, YogaConstants.AUTOf);
```
Comment (NickGerleman) [inline java/tests/generated/com/facebook/yoga/YGAbsolutePositionTest.java:84]: This doesn't seem valid

Comment (NickGerleman) [reply java/tests/generated/com/facebook/yoga/YGAbsolutePositionTest.java:84]: More generally, I think these tests assume JNI bindings for the Java Node exist to the new API.

Annoyingly, we don't run Java build or tests in GitHub Actions right now, but it is run when we import it into Phabricator, so it's usually pretty easy to pattern match an existing binding.

Comment (coado) [reply java/tests/generated/com/facebook/yoga/YGAbsolutePositionTest.java:84]: Okay, I can see that there is no `setPositionAuto` function for Java too.

Comment (NickGerleman) [reply java/tests/generated/com/facebook/yoga/YGAbsolutePositionTest.java:84]: Yep! That and generating `YogaConstants.AUTOf` (with the extraneous "f")

**Pull request closed without merging**